### DISCUSSION TITLE
Riverpod and Riverpod Hooks vs. Flutter Hooks

### DISCUSSION THREAD

**Discussion by vincent-herlemont** [rrousselGit/riverpod#2455] [Q&A] [answered]
Currenltly I use Flutter Hooks, I'm curious if Riverpod and Riverpod Hooks can effectively replace Flutter Hooks?

Could anyone with experience in using these libraries please share their thoughts on:

- Performance, ease of use, and maintainability.
- Any specific use cases where Flutter Hooks might be more suitable.

Thanks in advance for your input!

**Comment by rrousselGit** [comment, marked as answer]
Riverpod isn't meant to replace hooks. They are complementary

Providers are for globals state, and hooks for local state.

**Comment by masus04** [comment]
I am still somewhat confused as to what exactly the different flavours of riverpod are about.

What's the difference between `riverpod`, `flutter_riverpod` and `hooks_riverpod` and which one is recommended in which situation?

Is this something that could be described more clearly in the docs?

**Reply by hamza-imran75** [reply]
Hey the getting started page elaborates which to use when [riverpod.dev](https://riverpod.dev/docs/getting_started)

**Reply by jhigdon** [reply]
It doesn't seem to elaborate what @masus04 is asking, at least I can't find where it does;  @Hamza-Imran75 could you help point out where? Maybe I am just missing it too :(

**Reply by jhigdon** [reply]
@masus04 I just noticed the docs-v2 which seem to be much better https://docs-v2.riverpod.dev/docs/about_hooks

**Reply by hamza-imran75** [reply]
![image](https://github.com/rrousselGit/riverpod/assets/43512631/40e0b720-e052-4094-ae98-1848eef22eb5)

The first table in getting started page.

**Reply by masus04** [reply]
Yes, I saw the table, but are these packages implemented differently in the background? E.g. is `hooks_riverpod` using `Hooks` to implement the provider API? 

Are there any performance differences? 

Is `hooks_riverpod` simply packaging hooks with riverpod? If so what's the benefit over simply importing both?

@rrousselGit, could you elaborate?

**Reply by rrousselGit** [reply]
> Is hooks_riverpod simply packaging hooks with riverpod? If so what's the benefit over simply importing both?

Yep. It's doing nothing but that.  
There's no performance benefit.

The flutter_hooks page doc should explain the reasoning https://deploy-preview-2740--river-pod.netlify.app/docs/concepts/about_hooks

**Reply by rrousselGit** [reply]
There's also this page: https://deploy-preview-2740--river-pod.netlify.app/docs/essentials/first_request#flutter_hooks-considerations-combining-hookwidget-and-consumerwidget

**Reply by masus04** [reply]
Great, thank you for the link, you might want to link it near the above mentioned table, as it explains exactly what I was looking for :+1:

**Reply by rrousselGit** [reply]
There's no such table anymore in the new docs. Instead there's a toggle on the top-left of the website, with a "?" redirecting to that link
